### ISSUE TITLE
SpringFox docket add request parameter to specific end point

### ISSUE THREAD

**Issue by yuvalzi** [springfox/springfox#2129]
I am using springfox to generate swagger urls for rest end points. I have a specific rest endpoint that does not require a request parameter on the controller level, but requires request parameter on a filter that pre-handles the request.

I want to define a request parameter on the Docket in the Docket definition and not on the controller because annotating parameters on the controller makes it really dirty and hard to read.

In other words, i want to define swagger parameter without using ApiImplicitParams, and instead define it on the Docker.

Any advice? Thanks

**Comment by dilipkrish** [comment]
@yuvalzi you can use the `Docket#globalOperationParameters` and use `ParameterBuilder` to build parameters of your choice.

**Comment by yuvalzi** [comment]
Docket#globalOperationParameters is for all end-points. I am trying to configure a single end-point but i do not want to do it on the controller. It makes the controller hard to read

**Comment by dilipkrish** [comment]
Current that is not supported out of the box. There is a feature request #1910 for that
